File: 파이썬/4.큐/10845.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/10845
'''

import sys
input = sys.stdin.readline # 입력 속도를 빠르게 만들기 위해 꼭 쓴다.

n = int(input())
queue = []
for i in range(n) :
    a = input().split()
    if a[0] == 'push' : 
        queue.append(a[1])
    elif a[0] == 'pop' : 
        if len(queue) == 0 :
            print(-1)
        else :
            print(queue[0])
            queue.pop(0)
    elif a[0] == 'size' : 
        print(len(queue))
    elif a[0] == 'empty' :
        if len(queue) == 0 :
            print(1)
        else :
            print(0)
    elif a[0] == 'front' :
        if len(queue) == 0 :
            print(-1)
        else :
            print(queue[0])
    else :
        if len(queue) == 0 :
            print(-1) 
        else :
            print(queue[-1])

File: C++/10. 큐 - 초급/10845.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/10845
*/

#include <iostream>
#include <queue>

using namespace std;

int n, t;
string str;
queue<int> q;

int main(){
    
    cin >> n;
    
    for(int i=0; i<n; i++){
        cin >> str;
        if(str == "push"){
            cin >> t;
            q.push(t);
        }else if(str == "pop"){
            if(q.empty()){
                cout << "-1\n";
            }else{
                cout << q.front() << "\n";
                q.pop();
            }
        }else if(str == "size"){
            cout << q.size() << "\n";
        }else if(str == "empty"){
            cout << q.empty() << "\n";
        }else if(str == "front"){
            if(q.empty()){
                cout << "-1\n";
            }else{
                cout << q.front() << "\n";
            }
        }else{
            if(q.empty()){
                cout << "-1\n";
            }else{
                cout << q.back() << "\n";
            }
        }
    }
    
    return 0;
}